src/lib/kadm5/kadm_rpc.h

```diff
@@ -108,6 +108,15 @@ struct setkey3_arg {
 };
 typedef struct setkey3_arg setkey3_arg;
 
+struct setkey4_arg {
+	krb5_ui_4 api_version;
+	krb5_principal princ;
+	krb5_boolean keepold;
+	kadm5_key_data *key_data;
+	int n_key_data;
+};
```

Comment:
Another thought: don't we want to be able to add/delete keys for existing kvnos?  Heimdal can do this.  setkey4 seems like a good way to add keys for an existing keyset, though maybe not for deleting them.